internal/command/scale/count.go

```diff
@@ -92,30 +92,51 @@ func runScaleCount(ctx context.Context) error {
 	return runMachinesScaleCount(ctx, appName, appConfig, groups, maxPerRegion)
 }
 
-func parseGroupCounts(args []string, defaultGroupName string) (map[string]int, error) {
+func parseGroupCounts(ctx context.Context, args []string, defaultGroupName string) (map[string]int, error) {
 	groups := make(map[string]int)
 
+	var machineGroups map[string][]*fly.Machine
+
+	apply := func(group string, count string) error {
+		delta := 0
+		if strings.HasPrefix(count, "+") || strings.HasPrefix(count, "-") {
+			if machineGroups == nil {
+				flapsClient := flapsutil.ClientFromContext(ctx)
+				machines, _, err := flapsClient.ListFlyAppsMachines(ctx)
```

Comment:
this is a bit lame tbh, it needs two machine listings (one here, another in runMachineScaleCount) which is the most "expensive op"  and more probe to fail than any other call in this command.

I'd like to propose an alternative, return: `map[string]struct{absolute, relative int}`

and compute the expectedGroupCounts inside `runMachineScaleCount` after fetching the machine list.

if you feel brave, it can be moved inside `computeActions` where machines are already grouped by process group and absolute values can be computed easily; also tested.

Reply:
updated in d385475cba57a3be114b46599806045790bc6c50